src/index.js

```diff
@@ -140,6 +140,80 @@
+                            metadata
+                        })
+                        .then(() => {
+                            return vm.$pouch.putUser(username, metadata, db);
+                        })
+                        .catch(error => {
+                            return new Promise(resolve => {
+                                resolve(error);
+                            });
+                        });
+                },
+                deleteUser (username, db = databases[ defaultDB ]) {
+                    return db
+                        .deleteUser(username)
+                        .then(() => {
+                            return vm.$pouch.deleteUser(username, db);
+                        })
+                        .catch(error => {
+                            return new Promise(resolve => {
+                                resolve(error);
+                            });
+                        });
+                },
+                changePassword (username, password, db = databases[ defaultDB ]) {
+                    return db
+                        .changePassword(username, password)
+                        .then(() => {
+                            return vm.$pouch.changePassword(username, password, db);
+                        })
+                        .catch(error => {
+                            return new Promise(resolve => {
+                                resolve(error);
+                            });
+                        });
+                },
+                changeUsername (oldUsername, newUsername, db = databases[ defaultDB ]) {
+                    return db
+                        .changeUsername(oldUsername, newUsername)
+                        .then(() => {
+                            return vm.$pouch.changeUsername(oldUsername, newUsername, db);
+                        })
+                        .catch(error => {
+                            return new Promise(resolve => {
+                                resolve(error);
+                            });
+                        });
+                },
+                signUpAdmin (adminUsername, adminPassword, db = databases[ defaultDB ]) {
+                    return db
+                        .signUpAdmin(adminUsername, adminPassword)
+                        .then(() => {
+                            return vm.$pouch.signUpAdmin(adminUsername, adminPassword, db)
+                        })
+                        .catch(error => {
+                            return new Promise(resolve => {
+                                resolve(error);
+                            });
+                        });
+                },
+                deleteAdmin (adminUsername, db = databases[ defaultDB ]) {
+                    return db
+                        .deleteAdmin(adminUsername)
+                        .then(() => {
+                            return vm.$pouch.deleteAdmin(adminUsername, db)
```

Comment:
Not sure why you want to trigger the same instance function again when the promise is resolved. This is a double call in my opinion

Reply:
you're right, I don't know why, coding too fastly.